backend/coins/eth/account.go

```diff
@@ -564,10 +564,14 @@ func (account *Account) newTx(
 			}
 		}
 	}
-	gasLimit, err := account.coin.client.EstimateGas(context.TODO(), message)
-	if err != nil {
-		account.log.WithError(err).Error("Could not estimate the gas limit.")
-		return nil, errp.WithStack(errors.ErrInvalidData)
+	gasLimit := uint64(21000) // gas limit for standard ethereum transcations
+	if account.coin.erc20Token != nil {
+		var err error
+		gasLimit, err = account.coin.client.EstimateGas(context.TODO(), message)
```

Comment:
Can leave as is but just a couple notes how I would personally see it more in idiomatic Go:

```go
gasLimit := big.NewInt(21000)
if account.coin.erc20Token != nil {
  ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
  defer cancel()
  n, err := account.coin.client.EstimateGas(ctx, message)
  if err != nil {
    return nil, ...
  }
  gasLimit.SetUint64(n)
}
fee := new(big.Int).Mul(gasLimit, suggestedGasPrice)
```

- don't use `gasLimit` for possibly undefined values, even though the function returns early with an error, I know; it pays back over long term during refactoring
- use a context with a deadline; you never know what's on the other side

But maybe it's for a separate PR. I think there are few places like this.

Reply:
fixed first part

Reply:
thanks!